src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -1053,6 +1068,190 @@ private void AddPrimarySidClaim(List<Claim> instanceClaims)
                 safeAllocHandle.Dispose();
             }
         }
+
+        private void AddDeviceGroupSidClaims(List<Claim> instanceClaims, TokenInformationClass tokenInformationClass)
+        {
+            // special case the anonymous identity.
+            if (_safeTokenHandle.IsInvalid)
+                return;
+
+            SafeLocalAllocHandle safeAllocHandle = SafeLocalAllocHandle.InvalidHandle;
+            try
+            {
+                // Retrieve all group sids
+
+                safeAllocHandle = GetTokenInformation(_safeTokenHandle, tokenInformationClass);
```

Comment:
Any reason we do this in a manual try block rather than just using a using with it?

Reply:
Dunno, I just copied the code from netfx.